src/event-bridge/event-bus.ts

```diff
@@ -115,16 +120,16 @@ export interface IEventBusFilterable<E extends Event> {
   ): Rule<E, O>;
 }
 
-export interface IEventBus<E extends Event = Event>
+export interface IEventBus<in E extends Event = Event, O extends E = E>
```

Comment:
Might want to document the type parameters. Not immediately obvious what E and O mean

Reply:
In standard libraries, I usually prefer to use words for type parameters instead of symbols.

Reply:
Thanks, that was a helpful exercise. 

```ts
/**
 * @typeParam Evnt - the union type of events that this EventBus can accept.
 *                   `Evnt` is the covariant version of `OutEnvt` in that
 *                   the bus will accept any of `Evnt` while the EventBus can
 *                   emit any of `OutEvnt`.
 * @typeParam OutEvnt - the union type of events that this EventBus will emit through rules.
 *                      `OutEvnt` is the contravariant version of `Envt` in that
 *                      the bus will emit any of `OutEvnt` while the EventBus can
 *                      can accept any of `Evnt`. This type parameter should be left
 *                      empty to be inferred.
 */
```